modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -323,16 +326,38 @@ class QueueReader
 void rewindToStop(std::vector<Q*> &in_queues,
                   const std::size_t  this_id)
 {
-    for (auto &&qit : ade::util::indexed(in_queues))
-    {
-        auto id2 = ade::util::index(qit);
-        auto &q2 = ade::util::value(qit);
-        if (this_id == id2) continue;
+    // Will draining all queue & waiting for stop cmd for all queues except own queue
+    size_t expected_stop_count = std::count_if(in_queues.begin(), in_queues.end(), [] (const Q* ptr) {
+        return ptr != nullptr;
+    });
 
-        Cmd cmd;
-        while (q2 && !cv::util::holds_alternative<Stop>(cmd))
-            q2->pop(cmd);
+    // NB: extract 1 to skip own queue because stop cmd from own queue truggered rewind
+    expected_stop_count = expected_stop_count > 0 ? expected_stop_count - 1: expected_stop_count;
+    GAPI_LOG_INFO(nullptr, "id: " << this_id << ", queues count: " << in_queues.size() <<
+                           ", expected stop msg count: " << expected_stop_count);
+    size_t got_stop_count = 0;
+    while(got_stop_count < expected_stop_count) {
+        for (auto &&qit : ade::util::indexed(in_queues)) {
+            auto id2 = ade::util::index(qit);
+            auto &q2 = ade::util::value(qit);
+            if (this_id == id2) continue;
+
+            GAPI_LOG_INFO(nullptr, "drain next id: " << id2);
```

Comment:
this logs would be read with 
```
"rewindToStop - id: XXX, queues count: YYYY,  expected stop msg count: ZZZ"
"rewindToStop - drain next id: 0"
"rewindToStop - drain next id: 1"
"rewindToStop - got stop from id: 1 - (1/ZZZ)"
"rewindToStop - drain next id: 2"
"rewindToStop - got stop from id: 2 - (2/ZZZ)"
"rewindToStop - drain next id: 3"
...
"rewindToStop - drain next id: 0"
"rewindToStop - got stop from id: 0 - (ZZZ/ZZZ)"

```
together
and represent information about which queue is stopped but which is not yet
(It must have got actual `name` per each `queue` instead of id number in further logging improvements)